src/cmdstan/file.hpp

```diff
@@ -105,6 +105,51 @@ std::pair<std::string, std::string> get_basename_suffix(
   return {base, suffix};
 }
 
+/**
+ * Check if two file paths are the same file.
+ * @note This function only handles very basic access patterns.
+ *  It will miss cases such as
+ *  path1: ./a/b/../b/file.txt
+ *  path1: ./a/b/file.txt
+ * @param path1 first file path
+ * @param path2 second file path
+ * @return true if paths are the same file, false otherwise
+ */
+bool check_same_file(const std::string &path1, const std::string &path2) {
```

Comment:
Added some tests. I also changed the name to `check_approx_same_file` because imo it makes it more clear that we are not doing anything fancy to verify the path is 100% correct. (another reason to move up to c++17 imo)